src/Microsoft.Graph.Cli.Core/Http/LoggingHandler.cs

```diff
@@ -15,29 +16,42 @@ public class LoggingHandler : DelegatingHandler
     {
         if (Logger is null) return await base.SendAsync(request, cancellationToken);
 
-        var headers = request.Headers.Select((h) =>
-        {
-            var value = string.Join(",", h.Value);
-            if (h.Key == "Authorization")
-            {
-                value = "[PROTECTED]";
-            }
-            return string.Format("H:{0}={1}", h.Key, value);
-        }).Aggregate((a, b) => string.Join("\n\t", a, b));
-        Logger?.LogDebug("Calling:\n{0} {1}\n\t{2}", request.Method, request.RequestUri, headers);
+        Logger?.LogDebug("\nRequest:\n\n{0} {1} HTTP/{2}\n{3}\n{4}\n",
+            request.Method, request.RequestUri, request.Version,
+            HeadersToString(request.Headers),
+            await (request.Content?.ReadAsStringAsync() ?? Task.FromResult<string>(string.Empty))
+        );
+
         var response = await base.SendAsync(request, cancellationToken);
 
-        if (!response.IsSuccessStatusCode)
-        {
-            Logger?.LogDebug("{0}\t{1}\t{2}", request.RequestUri,
-                (int)response.StatusCode, response.Headers.Date);
-            Logger?.LogDebug("{0}", await response.Content.ReadAsStringAsync());
-        }
+        // If the response has a content length > 0 and is not a stream, get the content. Otherwise get the content length
```

Comment:
```suggestion
        // If the response has a content length > 0 and is a stream, get the content string. Otherwise get the content.
```
Any chance this is actually what you mean? The current comment is incongruent with the code in line 28.

Reply:
No. The condition is inverted which makes the comment confusing. We don't want to read an octet stream so this code returns the length if it's a stream and reads the content otherwise

Reply:
I'll expand the condition to make it easier to understand